README.md

```diff
@@ -77,6 +77,34 @@ Change the easing mechanism. This option takes effect when `duration` is set. In
 [bezier easing](https://www.npmjs.com/package/bezier-easing) similar to CSS
 [`cubic-bezier()`](<https://developer.mozilla.org/en-US/docs/Web/CSS/single-transition-timing-function#cubic-bezier()>).
 
+#### handleScroll(parent, {scrollLeft, scrollTop}, options)
+
+> Introduced in v1.4.0
+
+Type: `function`
+
+Take control over how the target is scrolled into view. This function is called
+for each parent node that need scrolling. `scrollLeft` and `scrollTop` are
+destination coordinates. The from coordinates you'll have to get yourself if you
+want to animate the transition using a different library.
+
+When using this option you likely don't need the built in animation feature. To
+cut down on filesize you can do the following adjustment if you are using a
+recent version of webpack or rollbar (and use ES6 imports):
+
+```diff
+-import scrollIntoViewIfNeeded from 'scroll-into-view-if-needed'
++import maybeScrollIntoView from 'scroll-into-view-if-needed/dist/calculate'
```

Comment:
If you add a file at the root of the project called `calculate` with the contents of:

```javascript
module.exports = require('./dist/calculate')
```

Then I could do:

```javascript
import maybeScrollIntoView from 'scroll-into-view-if-needed/calculate'
```

And that'd be a little nicer.

Or even better, if you expose an ESModules build (using something like rollup) then add a named export of `calculate` then I could do:

```javascript
import {calculate as maybeScrollIntoView} from 'scroll-into-view-if-needed'
```

And rollup's tree-shaking will remove any of the unused code when I run the downshift build :+1:

Thoughts?

Reply:
Great ideas! :D 

But regarding ESModules support, wouldn't the ./calculate.js file use

```
import calculate from './dist/calculate'
export {calculate}
```
For the tree shacking to work?
I didn't know it worked with `module.exports`?.

I do expose a ESModules build already using rollup :) The pkg.main field is pointing to the bundle, but the pkg.module (that webpack and browserify knows how to handle) is pointing to a tree-shake enabled build: ./dist/index.js 🙂 

If I move the code in src/index.ts to another file, and then import/export it with a default export in src/index.ts, and add a named export for `calculate`, this snippet should work:
```js
import {calculate as maybeScrollIntoView} from 'scroll-into-view-if-needed'
```

Am I right? Let's try it! 😄

Reply:
Ah, yeah, it looks like you've already got things set up nicely. So actually I'm pretty sure all you need to do is add a named export of the `calculate` function in `index.js` and that should do it. No need to move anything else. If you publish a beta, then I'll try it out on downshift :+1:

Reply:
Ok, so just FYI, I just tried:

```javascript
import calculate from 'scroll-into-view-if-needed/dist/calculate'
```

And jest failed to run because babel doesn't (and shouldn't) apply to `node_modules`. So here's what I suggest:

Create a `calculate/package.json` file where you have entries for `main` and `module` that point to a CJS version of the calculate file and the ESM version of the calculate file. Then users can simply do:

```javascript
import calculate from 'scroll-into-view-if-needed/calculate'
```

And it will work in node _and_ with tools like rollup/webpack.

Thoughts?

Reply:
Aha! I had no idea you could do that! :D I am out of town right now but I will circle back and will publish a beta with the new trick you just taught me and ping you :) Within the next 24 - 48 hours (because you never know with norwegian trains)

Reply:
No rush :)

Reply:
I will do some testing with this over the few next days and report back. I'll publish test packages on `scroll-into-view-if-needed@next` on npm